python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -0,0 +1,133 @@
+
+from sklearn.base import BaseEstimator
+from sklearn.model_selection import BaseCrossValidator
+
+# some nice properties of Sklearn models:
+# - they are always pickleable
+# - no logic is present in __init__
+
+ArrayType = Union[pd.DataFrame, np.ndarray]
+MetricType = Union[str, Callable[[BaseEstimator, ArrayType, ArrayType], float]]
+ScoringType = Union[MetricType, Iterable[MetricType], Dict[str, MetricType]]
+CVType = Union[int, Iterable, BaseCrossValidator]
+
+
+@PublicAPI(stability="alpha")
+class SklearnTrainer(Trainer):
+    """A Trainer for scikit-learn estimator training.
+
+    This Trainer runs the ``fit`` method of the given estimator in a
+    non-distributed manner on a single Ray Actor.
+
+    The ``n_jobs`` (or ``thread_count``) parameters will be set to match
+    the number of CPUs assigned to the Ray Actor.
+
+    Example:
+        .. code-block:: python
+
+            import ray
+
+            from ray.ml.train.integrations.sklearn import SklearnTrainer
+            from sklearn.ensemble import RandomForestRegressor
+
+            train_dataset = ray.data.from_items(
+                [{"x": x, "y": x + 1} for x in range(32)])
+            trainer = SklearnTrainer(
+                sklearn_estimator=RandomForestRegressor,
+                label_column="y",
+                scaling_config={
+                    "trainer_resources": {"CPU": 4}
+                },
+                datasets={"train": train_dataset}
+            )
+            result = trainer.fit()
+
+    Args:
+        estimator: A scikit-learn compatible estimator to use.
+        datasets: Ray Datasets to use for training and validation. Must include a
+            "train" key denoting the training dataset. If a ``preprocessor``
+            is provided and has not already been fit, it will be fit on the training
+            dataset. All datasets will be transformed by the ``preprocessor`` if
+            one is provided. All non-training datasets will be used as separate
+            validation sets, each reporting separate metrics.
+        label_column: Name of the label column. A column with this name
+            must be present in the training dataset.
+        scoring: Strategy to evaluate the performance of the model on
+            the validation sets and for cross-validation. Same as in
+            ``sklearn.model_selection.cross_validation``.
+            If ``scoring`` represents a single score, one can use:
+
+            - a single string;
+            - a callable that returns a single value.
+
+            If ``scoring`` represents multiple scores, one can use:
+
+            - a list or tuple of unique strings;
+            - a callable returning a dictionary where the keys are the metric
+            names and the values are the metric scores;
+            - a dictionary with metric names as keys and callables a values.
+
+        cv: Determines the cross-validation splitting strategy. If specified,
```

Comment:
How does this fit with the cross validation in Tuner cc @xwjiang2010?

Reply:
We can switch to any distributed CV we have later. ATM this will not be distributed